SumoLogic.Logging.Common/Sender/BufferFlushingTask.cs

```diff
@@ -118,32 +107,33 @@ protected BufferFlushingTask(BufferWithEviction<TBufferItem> messageQueue, ILog
         /// This flush and sends the messages.
         /// </summary>
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Exception unknown at compile time.")]
-        public Task Run()
+        public async Task Run()
         {
-            if (this.NeedsFlushing())
+            try
             {
-                this.IsFlushing = true;
-                try
-                {
-                    return this.FlushAndSend();
-                }
-                catch (Exception ex)
+                if (this.NeedsFlushing())
                 {
-                    if (this.Log.IsWarnEnabled)
-                    {
-                        this.Log.Warn("Exception while attempting to flush and send, Exception: " + ex.Message);
-                    }
+                    this.IsFlushing = true;
+                    await this.FlushAndSend().ConfigureAwait(false);
                 }
-            }
 
-            this.IsFlushing = false;
-            return CompletedTask;
+                this.IsFlushing = false;
+            }
+            catch (AggregateException ex)
+            {
+                var firstException = ex.Flatten().InnerExceptions?.FirstOrDefault() ?? ex.InnerException ?? ex;
+                Log.Warn($"HTTP Sender flush failed: {firstException.GetType()}: {firstException.Message}");
+            }

```

Comment:
When awaiting a task, then all exceptions are wrapped in AggregateException. And here we `await this.FlushAndSend()`

Reply:
Sorry you are right. When not using Task.Wait() but instead using await, then it will NOT throw AggregateException. Will fix the code to match new expectations.